components/pthread/include/esp_pthread.h

```diff
@@ -24,11 +27,33 @@ extern "C" {
 
 /** pthread configuration structure that influences pthread creation */
 typedef struct {
-    size_t stack_size;    ///< the stack size of the pthread
-    size_t prio;          ///< the thread's priority
-    bool inherit_cfg;     ///< inherit this configuration further
+    size_t stack_size;  ///< The stack size of the pthread
+    size_t prio;        ///< The thread's priority
+    bool inherit_cfg;   ///< Inherit this configuration further
+    // This member has to be an array and not a pointer due to the fact that the entire
+    // struct is copied by the thread when inherit_cfg is true which in turn likely
+    // will result in dangling pointers, especially if threads are created/destroyed
+    // dynamically at run time with different names etc.
+    char thread_name[configMAX_TASK_NAME_LEN + 1];  ///< The thread name.
+    int pin_to_core;    ///< The core id to pin the thread to. See components/pthread/KConfig for valid ranges.
```

Comment:
Suggest something like 'has the same meaning as xCoreId argument of xTaskCreatePinnedToCore' instead of pointing to Kconfig.

Reply:
Sure.